### PR TITLE
chore(#9636): update to node 22

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

#9636 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9636-node-22/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9636-node-22/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:9636-node-22/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,8 @@
-FROM alpine:3.19 AS base_build
+FROM node:22-alpine AS base_build
```
Comment (garethbowen) [inline api/Dockerfile:1]: This feels like a significant change. alpine haven't yet added node 22 to their packages, but this official node alpine package works fine. What do you think?

**Review by dianabarsan — APPROVED** [review]
NICE! I think we're good to go. We can always change back to `alpine:xx` later if we think that's the best image choice for us.

File: api/Dockerfile (hunk 1 above)
Comment (dianabarsan) [reply api/Dockerfile:1]: Oh, also, I saw that there are two 22 node alpine images, one based on alpine:3.19 and one based on alpine:3.20 https://github.com/nodejs/docker-node/tree/main/22
Which one are we getting?

Comment (dianabarsan) [reply api/Dockerfile:1]: The image is a bit more bloat (like auto adding yarn) in these images: https://github.com/nodejs/docker-node/blob/main/22/alpine3.20/Dockerfile

Comment (garethbowen) [reply api/Dockerfile:1]: Yeah I couldn't figure out why Node was adding yarn. That seems crazy...

Comment (garethbowen) [reply api/Dockerfile:1]: > Which one are we getting?

According to [the tags](https://hub.docker.com/_/node), `22-alpine` is the same as `22-alpine3.20`

Comment (garethbowen) [reply api/Dockerfile:1]: > It looks like alpine releases are really rare, but if a critical security fix gets pushed, we might need it asap.

Yeah you're right. It's a crazy circular dependency - if there's a critical flaw in Node then we need a node image to get it asap, but if there's a critical flaw in something else in alpine then we need the alpine image to get it asap.

We can always switch back to alpine if there's a flaw especially if they've published node 22 package.

Comment (dianabarsan) [reply api/Dockerfile:1]: agree.

**Pull request merged**